File: graph/linked-list/js/linkedlist.js
const Node = (value, nextNode = null) => ({
  value: value,
  nextNode: nextNode
});

// function to convert array to linked list
function fromList(alist) {
  if (!Array.isArray(alist)) {
    throw new TypeError('Argument must be of type array.');
  }

  if (!alist.length) {
    throw new Error('Array cannot be empty.');
  }

  if (alist.length === 1) {
    return Node(alist[0]);
  }

  return Node(alist[0], fromList(alist.slice(1)))
}

// function to convert linked list to array

function* traverse(node) {
  if (node) {
    yield node.value;
    yield* traverse(node.nextNode);
  }
}


(() => {
  const args = [3, 4, 5, 6, 7, 8, 9];
  const node = fromList(args);
  const traversedList = [...traverse(node)];
  console.log(traversedList);
})();

File: graph/linked-list/python/linkedlist.py

# class to represent a node
class Node:

  def __init__(self, value, next_node=None):
    self.value = value
    self.next_node = next_node


  def __repr__(self):
    return 'Node({})'.format(self.value)



# a function to create a linked list from the list
def from_list(alist=[]):
  if not alist:
    raise ValueError('List cannot be empty.')
  if len(alist) == 1:
    return Node(alist[0])
  return Node(alist[0], next_node=from_list(alist[1:]))

# a function to traverse a linked list
def traverse(node):
  if node:
    yield node.value
    yield from traverse(node.next_node)




if __name__ == '__main__':
  alist = list(range(10))
  node = from_list(alist)
  traversed_list = list(traverse(node))
  print(traversed_list)